File: microspice/error/inp_error.py
from microspice.error.error import Error

class InputError(Error):
    def __init__(self, inp_file, inp_line, inp_msg):
        super().__init__()
        self.error_type = "Input error"
        self.file_name  = inp_file
        self.line_num   = inp_line
        self.message    = inp_msg
    
    def get_message(self):
        return f"{self.file_name} line {self.line_num} : {self.message}"

class NumArgsError(InputError):
    def __init__(self, inp_file, inp_line, inp_act, inp_req):
        super().__init__(inp_file, inp_line, f"Mismatch in number of arguments ({inp_act} given, {inp_req} required)")

File: microspice_matlab/+error/+inp_error/inp_error.m
classdef inp_error < error.error
    properties
        file_name   (1, 1)  string;     % The name of file where error was detected
        line_num    (1, 1)  int32;      % The line number of the input file at which error was detected
        message     (1, 1)  string;     % The message to be encoded
    end
    
    methods
        function obj = inp_error(inp_file, inp_line, inp_msg)
            obj.error_type = "Input error";
            obj.file_name = inp_file;
            obj.line_num  = inp_line;
            obj.message   = inp_msg;
        end
        
        function ret = get_message(obj)
            ret = obj.file_name + " line " + obj.line_num + " : " + obj.message;
        end
    end
end
